### PR TITLE
[DO NOT MERGE] Add Lua 5.4 support

### PR DESCRIPTION
**Pull request by Tieske**
Experimental;

* added windows binaries to gitignore (`*.dll`)
* removed uper version constrained from the SCM rockspec
* added Lua 5.4 to the test matrix
* added `llimits.h` and `lobject.h`, plain copies from Lua 5.4 source tree

### CODE REVIEW

File: cluacov-scm-1.rockspec [hunk 1]
```diff
@@ -13,7 +13,7 @@ number of lines incorrectly marked as missed. Depends on LuaCov.
    license = "MIT"
 }
 dependencies = {
-   "lua >= 5.1, < 5.4",
+   "lua >= 5.1",
```
Comment (daurnimator) [inline cluacov-scm-1.rockspec:16]: I think the maximum version was suggested by @hishamhm  ?

**Comment by Tieske** [comment]
this is non-functional, it's a quick experiment to add support. My guess is that the source files coming in from the Lua source are not to be copied verbatim, but need changes added to them.

A diff between the Lua 5.1/2/3 and their Lua originals would probably shed some light on that

File: cluacov-scm-1.rockspec (hunk 1 above)
Comment (Tieske) [reply cluacov-scm-1.rockspec:16]: Since this is the SCM rockspec, I think the limit should not be there. In the versioned ones, it should be there.

**Comment by alerque** [comment]
If a PR isn't ready for review and merging you can post it as a Draft which will automatically block merging. This is an option when you create the PR or this existing ones can be converted using the link under the reviewers section in the sidebar.

**Comment by hishamhm** [comment]
Superseded by #2!

**Pull request closed without merging**